File: react/employee-management-ui/src/services/employeeService.js
import api from "../api/api";

export const getEmployees = () => {
    return api.get("/api/employees");
};

export const addEmployee = (employee) => {
    return api.post("/api/employees", employee);
};

export const getEmployeeById = (id) => {
    return api.get(`/api/employees/${id}`);
};

export const updateEmployee = (id, employee) => {
    return api.put(`/api/employees/${id}`, employee);
};

export const deleteEmployee = (id) => {
    return api.delete(`/api/employees/${id}`);
};

File: Employee Management/src/main/java/com/rishabh/employeemanagement/service/EmployeeService.java
package com.rishabh.employeemanagement.service;

import com.rishabh.employeemanagement.model.Employee;
import java.util.List;

public interface EmployeeService {

    Employee createEmployee(Employee employee);

    Employee getEmployeeById(Long id);

    List<Employee> getAllEmployees();

    Employee updateEmployee(Long id, Employee employee);

    void deleteEmployee(Long id);
}